lib/rules/prefer-find-by.ts

```diff
@@ -0,0 +1,88 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import {
+  isIdentifier,
+  isCallExpression,
+  isMemberExpression,
+  isArrowFunctionExpression,
+} from '../node-utils';
+import { getDocsUrl, SYNC_QUERIES_COMBINATIONS } from '../utils';
+
+export const RULE_NAME = 'prefer-find-by';
+
+type Options = [];
+export type MessageIds = 'preferFindBy';
+// TODO check if this should be under utils.ts - there are some async utils
+export const WAIT_METHODS = ['waitFor', 'waitForElement', 'wait']
+
+export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Suggest using find* instead of waitFor to wait for elements',
+      category: 'Best Practices',
+      recommended: false,
```

Comment:
Why not recommend this one?

Reply:
I think it should be recommended, indeed.

Reply:
"warn" or "error" ?

Reply:
Nice question. I think warn is enough? This rules doesn't prevent any malfunctioning code, just warns about a preferred way of writing it.

Reply:
I added `warn`